### PR TITLE
feat: Implements the requirer side of the ingress integration

### PR DESCRIPTION
**Pull request by saltiyazan**
# Description

- Adds the ingress and cert-transfer integrations
- Fetches the libs
- Adds integration test

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by kayra1 — DISMISSED** [review]

**Comment by saltiyazan** [comment]
This won't work if we merge it right away, it needs the tls integrations to be merged first, I'll make it a draft until then.

File: tests/integration/test_charm.py [hunk 1]
```diff
@@ -30,6 +30,7 @@
 
 LOKI_APPLICATION_NAME = "loki-k8s"
 PROMETHEUS_APPLICATION_NAME = "prometheus-k8s"
+TRAEIK_K8S_APPLICATION_NAME = "traefik-k8s"
```
Comment (ghislainbourgeois) [inline tests/integration/test_charm.py:33]:
```suggestion
TRAEFIK_K8S_APPLICATION_NAME = "traefik-k8s"
```

File: tests/integration/test_charm.py [hunk 2]
```diff
@@ -63,6 +64,12 @@ async def test_build_and_deploy(ops_test: OpsTest, request: pytest.FixtureReques
     await ops_test.model.deploy(
         "loki-k8s", application_name=LOKI_APPLICATION_NAME, trust=True, channel="stable"
     )
+    await ops_test.model.deploy(
+        TRAEIK_K8S_APPLICATION_NAME,
+        application_name=TRAEIK_K8S_APPLICATION_NAME,
```
Comment (ghislainbourgeois) [inline tests/integration/test_charm.py:69]:
```suggestion
        TRAEFIK_K8S_APPLICATION_NAME,
        application_name=TRAEFIK_K8S_APPLICATION_NAME,
```

File: tests/integration/test_charm.py [hunk 3]
```diff
@@ -187,13 +194,52 @@ async def test_given_loki_and_prometheus_related_to_notary_all_charm_statuses_ac
...
+        relation1=f"{TLS_PROVIDER_APPLICATION_NAME}:certificates",
+        relation2=f"{TRAEIK_K8S_APPLICATION_NAME}",
+    )
+    await ops_test.model.integrate(
+        relation1=f"{TLS_PROVIDER_APPLICATION_NAME}:certificates",
+        relation2=f"{APP_NAME}:access-certificates",
+    )
+    await ops_test.model.integrate(
+        relation1=f"{APP_NAME}:ingress",
+        relation2=f"{TRAEIK_K8S_APPLICATION_NAME}:ingress",
```
Comment (ghislainbourgeois) [inline tests/integration/test_charm.py:217]:
```suggestion
        relation2=f"{TRAEFIK_K8S_APPLICATION_NAME}",
    )
    await ops_test.model.integrate(
        relation1=f"{TLS_PROVIDER_APPLICATION_NAME}:certificates",
        relation2=f"{APP_NAME}:access-certificates",
    )
    await ops_test.model.integrate(
        relation1=f"{APP_NAME}:ingress",
        relation2=f"{TRAEFIK_K8S_APPLICATION_NAME}:ingress",
```

File: tests/integration/test_charm.py [hunk 4]
```diff
@@ -187,13 +194,52 @@ async def test_given_loki_and_prometheus_related_to_notary_all_charm_statuses_ac
...
+    )
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME, TRAEIK_K8S_APPLICATION_NAME],
```
Comment (ghislainbourgeois) [inline tests/integration/test_charm.py:220]:
```suggestion
        apps=[APP_NAME, TRAEFIK_K8S_APPLICATION_NAME],
```

File: tests/integration/test_charm.py [hunk 5]
```diff
@@ -221,6 +267,14 @@ async def run_get_certificate_action(ops_test: OpsTest) -> str:
     return action_output.get("certificates", "")
 
 
+async def run_show_traefik_proxied_endpoints_action(ops_test: OpsTest) -> str:
+    assert ops_test.model
+    traefik_k8s_unit = ops_test.model.units[f"{TRAEIK_K8S_APPLICATION_NAME}/0"]
```
Comment (ghislainbourgeois) [inline tests/integration/test_charm.py:272]:
```suggestion
    traefik_k8s_unit = ops_test.model.units[f"{TRAEFIK_K8S_APPLICATION_NAME}/0"]
```

**Review by ghislainbourgeois — APPROVED** [review]
Approved, assuming the small typo fixes suggested are included.

**Pull request merged**